packages/ui-react/src/containers/AccountModal/forms/Login.tsx

```diff
@@ -29,13 +29,21 @@ const Login: React.FC<Props> = ({ messageKey }: Props) => {
 
   const queryClient = useQueryClient();
 
+  // temporary state to show success message
+  const [successState, setSuccessState] = React.useState(false);
+
   const loginSubmitHandler: UseFormOnSubmitHandler<LoginFormData> = async (formData, { setErrors, setSubmitting, setValue }) => {
     try {
+      setSubmitting(true);
       await accountController.login(formData.email, formData.password, window.location.href);
       await queryClient.invalidateQueries(['listProfiles']);
 
-      // close modal
-      navigate(modalURLFromLocation(location, null));
+      // close the modal after successful login
+      setSuccessState(true);
+      setTimeout(() => {
+        navigate(modalURLFromLocation(location, null));
+        setSuccessState(false);
+      }, 1000);
```

Comment:
I'm not sure if we should delay closing the modal to announce the success message. This change has a big impact on sighted users who don't see the success message at all. 

Currently, for sighted users, the form loading state (submitting) is reset to false, but the modal is kept open for 1s. There is no clear indicator of what is happening. 

I think (correct me if I'm wrong) that we need this mechanism for more forms. Maybe we can think of a more generic solution when the UI doesn't show feedback or the UI is removed immediately after completion.

The first thing that comes to mind is a bit more complex but can work for many more use cases. We can do this the React way or choose to do it vanilla.

```tsx
type Variant = 'success' | 'error' | 'warning' | 'info';
type Announcement = { message: string; variant: Variant };
type AnnounceFn = (message: string, variant: Variant, duration?: number) => void;
type ContextValue = { announce: AnnounceFn };

const AriaAnnouncerContext = createContext<ContextValue | undefined>(undefined);

// Hook for easy access to announce fn

export const useAriaAnnouncer = () => {
  const announcer = useContext(AriaAnnouncerContext);

  if (!announcer) throw new Error('Announcer context is not defined');

  return announcer.announce;
};

// Provider used in App.tsx

export const AriaAnnouncerProvider = ({ children }: PropsWithChildren) => {
  const [announcement, setAnnouncement] = useState<Announcement | null>(null);

  const announce = useCallback((message: string, variant: Variant, duration = 1000) => {
    setAnnouncement({ message, variant });

    setTimeout(() => {
      setAnnouncement(null);
    }, duration);
  }, []);

  return (
    <AriaAnnouncerContext.Provider value={{ announce }}>
      {announcement && (
        <FormFeedback variant={announcement.variant} visible={false}>
          {announcement.message}
        </FormFeedback>
      )}
      {children}
    </AriaAnnouncerContext.Provider>
  );
};

// Usage in Form

// temporary state to show success message
const announce = useAriaAnnouncer();

const loginSubmitHandler: UseFormOnSubmitHandler<LoginFormData> = async (formData, { setErrors, setSubmitting, setValue }) => {
  try {
    await accountController.login(formData.email, formData.password, window.location.href);
    await queryClient.invalidateQueries(['listProfiles']);

    announce(t('login.sign_in_success'), 'success');

    navigate(modalURLFromLocation(location, null));
    
    // ....

```

Let me know what you think 😄

Reply:
Hi @Christiaan,

Thank you for the insightful proposal regarding the AriaAnnouncer implementation. I agree that integrating it with our current modal behavior could greatly improve user experience. 

My suggestion is to embed the provider within the forms directly. This approach will not only streamline the reuse of logic across various forms but also enhance the visual cues for sighted users. Specifically, it will facilitate the display of both error messages and non-visible, spoken success announcements within the specific form or container where it's employed.

Reply:
As I argued in my previous comment, I would suggest removing the redirect on timeout ;-)

Reply:
I was hesitant to suggest making it visible as well, but that's too much for this moment. Perhaps we could implement a simple snackbar later (next sprint?). 

I agree to choose an alternative to the delay to close the modal. Although I find the need for a provider and hook just for an aria div a bit much. You could alternatively do it vanilla and create a util, although I'm not sure (might need some tweaking, just the idea):

```typescript
type Variant = 'polite' | 'assertive' | 'warning' | 'error';

const announceMessage = (message: string, variant: Variant, duration = 2000) => {
    const ariaAnnouncement = document.createElement('div');
    ariaAnnouncement.style.position = 'absolute';
    ariaAnnouncement.style.left = '-100px';
    ariaAnnouncement.style.width = '1px';
    ariaAnnouncement.style.height = '1px';
    ariaAnnouncement.style.margin = '-1px';
    ariaAnnouncement.style.overflow = 'hidden';
    ariaAnnouncement.setAttribute('aria-live', ariaLiveValues[variant] || 'polite');
    ariaAnnouncement.setAttribute('role', 'alert');
    ariaAnnouncement.textContent = message;

    document.body.appendChild(ariaAnnouncement);

    setTimeout(() => {
        document.body.removeChild(ariaAnnouncement);
    }, duration);
}
```

Reply:
I'm leaning towards @ChristiaanScheermeijer's method as it offers a scalable solution that can integrate with global notifications or custom visual components in the future. This approach minimizes the need for extensive refactoring later on.

For instance, we could expand the `AriaAnnouncer` to include visual notifications, by adding a component like `VisualNotifier` in the `AriaAnnouncerProvider`. This component can be styled and positioned as needed.